### PR TITLE
IGVF-1722-analysis-step-version-schema

### PR DESCRIPTION
**Pull request by khzlin**

### CODE REVIEW

File: src/igvfd/schemas/analysis_step_version.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            "shared_dependencies.json#/standard_status"
+        ]
+    },
+    "properties": {
+        "schema_version": {
+            "default": "1"
+        },
+        "analysis_step": {
+            "title": "Analysis Step",
+            "description": "The analysis step in which this version belongs to.",
```
Comment (jenjou) [inline src/igvfd/schemas/analysis_step_version.json:44]: This sentence is confusing to me, maybe "The analysis step which this version belongs to"?

File: src/igvfd/schemas/changelogs/alignment_file.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 ## Changelog for *`alignment_file.json`*
 
+## Minor changes since schema version 10
+* Add `analysis_step_version`.
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/alignment_file.md:4]: Blank line between lines 3 and 4

File: src/igvfd/schemas/changelogs/configuration_file.md [hunk 3]
```diff
@@ -6,6 +6,7 @@
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.2`.
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.3`.
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.4`.
+* Add `analysis_step_version`.
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/configuration_file.md:9]: New change should be at the top of the section (before line 5 here)

File: src/igvfd/schemas/changelogs/tabular_file.md [hunk 4]
```diff
@@ -2,9 +2,13 @@
 
 ### Minor changes since schema version 11
 
+<<<<<<< HEAD
 * Extend `content_type` enum list to include `differential gene expression quantifications`.
 * Extend `content_type` enum list to include `differential transcript expression quantifications`.
 * Extend `content_type` enum list to include `MPRA sequence designs`.
+=======
+* Add `analysis_step_version`.
+>>>>>>> 9b82ec5f (IGVF-1722-analysis-step-version-schema)
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/tabular_file.md:11]: Looks like this file was messed up during rebase

File: src/igvfd/schemas/file.json [hunk 5]
```diff
@@ -230,6 +230,16 @@
         "accession": {
             "accessionType": "FI"
         },
+        "analysis_step_version": {
+            "title": "Analysis step version",
```
Comment (jenjou) [inline src/igvfd/schemas/file.json:234]: Title should be capitalized Analysis Step Version

File: src/igvfd/tests/test_schema_sequence_file.py [hunk 6]
```diff
@@ -380,3 +380,14 @@ def test_controlled_sequence_file_release(testapp, controlled_sequence_file_obje
         expect_errors=True
     )
     assert res.status_code == 200
+
+
+def test_analysis_step_version(testapp, sequence_file, analysis_step_version):
```
Comment (jenjou) [inline src/igvfd/tests/test_schema_sequence_file.py:385]: Somewhat confused what this test is testing for -- just making sure that the property can be added to a file? If so, I think usually the inserts already cover making sure that the property can be used so we might not need this test.

File: src/igvfd/schemas/analysis_step_version.json (hunk 1 above)
Comment (khzlin) [reply src/igvfd/schemas/analysis_step_version.json:44]: done

File: src/igvfd/schemas/changelogs/alignment_file.md (hunk 2 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/alignment_file.md:4]: done

File: src/igvfd/schemas/changelogs/configuration_file.md (hunk 3 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/configuration_file.md:9]: done

File: src/igvfd/schemas/changelogs/tabular_file.md (hunk 4 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/tabular_file.md:11]: done

File: src/igvfd/schemas/file.json (hunk 5 above)
Comment (khzlin) [reply src/igvfd/schemas/file.json:234]: done

File: src/igvfd/tests/test_schema_sequence_file.py (hunk 6 above)
Comment (khzlin) [reply src/igvfd/tests/test_schema_sequence_file.py:385]: ok, removed

File: src/igvfd/schemas/changelogs/matrix_file.md [hunk 7]
```diff
@@ -7,6 +7,7 @@
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.2`.
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.3`.
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.4`.
+* Add `analysis_step_version`.
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/matrix_file.md:10]: Add new entry at top of section

File: src/igvfd/schemas/changelogs/reference_file.md [hunk 8]
```diff
@@ -3,6 +3,7 @@
 ### Minor changes since schema version 14
 
 * Add calculated property `input_file_for`.
+* Add `analysis_step_version`.
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/reference_file.md:6]: Add new entry at top of section

File: src/igvfd/schemas/changelogs/genome_browser_annotation_file.md [hunk 9]
```diff
@@ -8,6 +8,7 @@
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.2`.
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.3`.
 * Extend `collections` enum list to include `IGVF_catalog_beta_v0.4`.
+* Add `analysis_step_version`.
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/genome_browser_annotation_file.md:11]: Add new entry at top of section

File: src/igvfd/schemas/changelogs/sequence_file.md [hunk 10]
```diff
@@ -3,6 +3,7 @@
 ### Minor changes since schema version 14
 
 * Add calculated property `input_file_for`.
+* Add `analysis_step_version`.
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/sequence_file.md:6]: Add new entry at top of section

File: src/igvfd/schemas/changelogs/signal_file.md [hunk 11]
```diff
@@ -3,6 +3,7 @@
 ### Minor changes since schema version 9
 
 * Add calculated property `input_file_for`.
+* Add `analysis_step_version`.
```
Comment (jenjou) [inline src/igvfd/schemas/changelogs/signal_file.md:6]: Add new entry at top of section

File: src/igvfd/schemas/changelogs/matrix_file.md (hunk 7 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/matrix_file.md:10]: done

File: src/igvfd/schemas/changelogs/reference_file.md (hunk 8 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/reference_file.md:6]: done

File: src/igvfd/schemas/changelogs/genome_browser_annotation_file.md (hunk 9 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/genome_browser_annotation_file.md:11]: done

File: src/igvfd/schemas/changelogs/sequence_file.md (hunk 10 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/sequence_file.md:6]: done

File: src/igvfd/schemas/changelogs/signal_file.md (hunk 11 above)
Comment (khzlin) [reply src/igvfd/schemas/changelogs/signal_file.md:6]: done

**Review by jenjou — DISMISSED** [review]

**Review by jenjou — COMMENTED** [review]
Missed it earlier but Ian's comment has reminded me, there is also Model File which needs changelog update.

**Review by jenjou — APPROVED** [review]

**Pull request merged**